src/main/java/de/dennisguse/opentracks/content/sensor/SensorDataCycling.java

```diff
@@ -59,17 +59,29 @@ public int getCrankRevolutionsTime() {
             return crankRevolutionsTime;
         }
 
+        @NonNull
+        @Override
+        protected Float getNoneValue() {
+            return 0f;
+        }
+
         public void compute(Cadence previous) {
             if (hasData() && previous != null && previous.hasData()) {
                 float timeDiff_ms = UintUtils.diff(crankRevolutionsTime, previous.crankRevolutionsTime, UintUtils.UINT16_MAX) / 1024f * UnitConversions.S_TO_MS;
                 if (timeDiff_ms <= 0) {
                     Log.e(TAG, "Timestamps difference is invalid: cannot compute cadence.");
                     value = null;
-                } else {
```

Comment:
I am not sure about this.
On the one hand, the documentation says that the value can overflow and should be treated like this.
On the other hand, we would only miss one rotation if this happens - so the impact is not that big.

The question in my head actually is more: why does the sensor behaves like it does?
Could it be that the sensor turns off (e.g., power saving) and then resets the counter?
And could we detect this in the BluetoothManager (e.g., does a disconnect happen)?

Reply:
Really good questions.

>  Could it be that the sensor turns off (e.g., power saving) and then resets the counter?

I'm not sure but according the tests it looks that sensor turns off and it restarts counter after a long period without activity. But we should look into this.

>  And could we detect this in the BluetoothManager (e.g., does a disconnect happen)?

That will be awesome and we can try to accomplish it.

For now, I think is better missing a rotation that saving a huge value for cadence or speed that would throws off statistics. What do you think? Should we accept this workaround for now?